ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -229,6 +229,109 @@ def test_delete_with_nonexistent_hostgroup(self, automember_hostgroup,
                 reason=u'%s: Automember rule not found' % hostgroup1.cn)):
             command()
 
+    def test_create_with_existent_group(self, automember_group, group1):
+        """ Try to add a rule that already exists """
+        group1.ensure_exists()
+        command = automember_group.make_create_command()
+        with raises_exact(errors.NotFound(
```

Comment:
`NotFound` error doesn't make sense here, since you want to use existing group. You may want to use `DuplicateEntry` instead (check `ipalib/errors.py` for all available)

Reply:
To ensure that the exception DuplicateEntry is returned, the code needs to call automembergroup.ensure_exists() first (right after group1.ensure_exists()).